packages/@react-types/shared/src/dnd.d.ts

```diff
@@ -117,7 +117,7 @@ export interface ItemDropTarget {
   /** The drop target type. */
   type: 'item',
   /** The item key. */
-  key: Key,
+  key?: Key | null,
```

Comment:
When would the item not have a key? This seems to be causing a lot of unnecessary null checking in our examples. Seems like if you're dropping on/between items, there should always be a key.